tools/config/__init__.py

```diff
@@ -1058,9 +1063,79 @@ def validate_config(self):
 
         Arguments: None
         """
+
         params, _ = self.get_config_data()
+        err_msg = ""
+
+        for name, param in sorted(params.items()):
+            min      = param.value_min
+            max      = param.value_max
+            accepted = param.accepted_values
+            value    = param.value
+
+            # Config parameters that are only defined but do not have a default
+            # value should not be range limited
+            if value is not None:
+                if (min is not None or max is not None) and (accepted is not None):
+                    err_msg += "\n%s has both a range and list of accepted values specified. Please only "\
+                                "specify either value_min and/or value_max, or accepted_values"\
+                                    % param
+                else:
+                    if re.match(r'^(0[xX])[A-Fa-f0-9]+$|^[0-9]+$', str(value)):
+                        # Value is a hexadecimal or numerical string value
+                        # Convert to a python integer and range check/compare to
+                        # accepted list accordingly
+                        if re.match(r'^(0[xX])[A-Fa-f0-9]+$', str(value)):
```

Comment:
Heads up, `int()` can actually cope with `0x` prefixed hex strings! https://docs.python.org/2/library/functions.html#int
(also stackoverflow example: https://stackoverflow.com/questions/209513/convert-hex-string-to-int-in-python)

You can reduce your implementation by one regular expression if you like, which is always a good thing 😄